### PR TITLE
feat: support goerli

### PR DESCRIPTION
**Pull request by md0x**
Changes proposed in this PR:
- Add support for goerli
- Update readme env variables

### CODE REVIEW

**Review by mrice32 — APPROVED** [review]
LGTM!

File: README.md [hunk 1]
```diff
@@ -33,13 +33,31 @@ yarn
...
-PROVIDER_URL=https://mainnet.infura.io/v3/abcdef // Ethereum mainnet RPC provider.
+SENDER_PRIVATE_KEY=<your_sender_private_key>  # Private key of the actor authorized to call unlockLatestValue on the Oval.
+AUTH_PRIVATE_KEY=<your_auth_private_key>      # Root private key for deriving searcher-specific keys for signing bundles.
+PROVIDER_URL=<your_provider_url>              # Ethereum mainnet/goerli RPC provider URL.
+BUILDERS=<builders_json_array>                # JSON array specifying the builders for MEV-Share.
+CHAIN_ID=<network_chain_id>                   # Chain ID of the Ethereum network you are targeting, 1 or 5 for mainnet or goerli.
+OVAL_CONFIGS=<oval_configs_json>              # JSON string that maps Oval contract addresses to their specific configurations.
 ```
 
+`OVAL_CONFIGS` is a JSON string that maps Oval contract addresses to their specific configurations. Each entry in this JSON object should have the following format:
```
Comment (daywiss) [inline README.md:44]: is this an array? if so its not clear in the example

File: src/lib/helpers.ts [hunk 2]
```diff
@@ -247,3 +248,17 @@ export function getPrivateKey(input: string): string {
   if (!isHexString(privateKey, 32)) throw new Error(`Value ${input} not a valid private key`);
   return privateKey;
 }
+// Calculate the maximum block number to target with bundles by chainId.
+// In mainnet this is always the targetBlock, but in Goerli we add 24 blocks to the targetBlock.
+export function getMaxBlockByChainId(chainId: number, targetBlock: number) {
```
Comment (daywiss) [inline src/lib/helpers.ts:253]: id suggest doign some kind of mapping here, even though we arent supporting more than 2, its just easier to understand

```
  const additionalBlocks = {
   [MAINNET_CHAINID]:0
    [GOERLI_CHAIN_ID]:24
 }
```
then have your function just reference into the object to calc the offset, throw an error if an unsupported chain is used

**Review by daywiss — APPROVED** [review]
couple small things

File: README.md [hunk 3]
```diff
@@ -33,13 +33,31 @@ yarn
 Define a `.env` file and define, at minimum, the following (see [here](./src/lib/env.ts) for more settings):
 
 ```
-OVAL_ADDRESS=0x420 // The address of the Oval contract you want to target with the Oval Node.
-SENDER_PRIVATE_KEY=0x6969 // The private key of the permissioned actor who can call unlockLatestValue on the associated Oval.
-AUTH_PRIVATE_KEY=0x6666 // The root private key used to derive searcher specific keys for signing bundles.
-REFUND_ADDRESS=0x42069 // The refund address you want to send the OEV kickback to.
-PROVIDER_URL=https://mainnet.infura.io/v3/abcdef // Ethereum mainnet RPC provider.
+SENDER_PRIVATE_KEY=<your_sender_private_key>  # Private key of the actor authorized to call unlockLatestValue on the Oval.
```
Comment (Reinis-FRP) [inline README.md:36]: If using `OVAL_CONFIGS` then there is no need for `OVAL_ADDRESS`, `SENDER_PRIVATE_KEY`, `REFUND_ADDRESS` and `REFUND_PERCENT` - these are here only for backwards compatibility when serving one Oval contract instance. So either we don't document any of them or document all 4 explaining you don't need `OVAL_CONFIGS` then.

File: README.md (hunk 1 above)
Comment (Reinis-FRP) [reply README.md:44]: No, Its JSON object where keys are Oval contract addresses mapping to nested config object

File: src/lib/helpers.ts (hunk 2 above)
Comment (Reinis-FRP) [reply src/lib/helpers.ts:253]: agree, and also better move all these chain ids and their additional target block values to constants leaving only the logic here

File: src/lib/env.ts [hunk 4]
```diff
@@ -26,10 +26,16 @@ try {
 } catch {
   stringifiedFallbackOvalConfigs = undefined;
 }
+// Chain ID and network
+const chainId = getInt(getEnvVar("CHAIN_ID", fallback.chainId.toString()));
+if (chainId !== 1 && chainId !== 5) throw new Error(`Unsupported chainId: ${chainId}`);
+const network: "mainnet" | "goerli" = chainId === 1 ? "mainnet" : "goerli";
```
Comment (Reinis-FRP) [inline src/lib/env.ts:32]: Better store only chainId in env and move network name to constants mapping

File: src/lib/env.ts [hunk 5]
```diff
@@ -26,10 +26,16 @@ try {
 } catch {
   stringifiedFallbackOvalConfigs = undefined;
 }
+// Chain ID and network
+const chainId = getInt(getEnvVar("CHAIN_ID", fallback.chainId.toString()));
+if (chainId !== 1 && chainId !== 5) throw new Error(`Unsupported chainId: ${chainId}`);
```
Comment (Reinis-FRP) [inline src/lib/env.ts:31]: Better check against mapping in constants if we support the chain. If testing on Sepolia this would start getting too long here

**Review by Reinis-FRP — APPROVED** [review]
looking good!

**Pull request merged**